### PR TITLE
Bugfix of slowdata coming from network being stripped when in gateway mode

### PR DESCRIPTION
**Pull request by F4FXL**
Hi,

I think I fixed it, but not sure if there any side effects. It seems like the code was so made that the slow data sync bytes were not set when in gateway mode.
I commented those lines of code, see around line 1650

73
Geoffrey F4FXL

### CODE REVIEW

**Comment by dl5di** [comment]
Hi Geoffrey!
This is not my code, I only moved it from the former BerliOS repostitory 
to Github when Jonathan was moving and not available for a longer time.
I need a repository to feed the package builders for the Linux 
installation packages.
But the code is still under Jonathan's control.

I forwarded your email to Jonathan.
Unfortunately he currently only has very limited internet access and it 
is not easy to discuss it with him more in detail.
Please see his answer here:

/RE: [OpenDV] bugfix for gateway mode stripping slow data, not testedyet 
(#9)//
//Mon, 10 Aug 2015 13:08:28 +0100//
//Jonathan Naylor <naylorjs@yahoo.com>//
//
//Hi Hans//
//
//The slow data is stripped for a reason, if the header is missed, 
re-sync will occur.
The original slow data header will not allow the data to be passed, so 
it needs to be replaced with a suitable slow data header.//
//
//If I remove my slow data modification then reliability will be lost.
Is this change worth that? If so, then I'll change it.//
//
//Jonathan. G4KLX//
/

73 de Hans DL5DI

Am 13.08.2015 um 20:41 schrieb Geoffrey Merck:
>
> Hi,
>
> I think I fixed it, but not sure if there any side effects. It seems 
> like the code was so made that the slow data sync bytes were not set 
> when in gateway mode.
> I commented those lines of code, see around line 1650
>
> 73
> Geoffrey F4FXL
>
> ------------------------------------------------------------------------
>
>
>         You can view, comment on, or merge this pull request online at:
>
> https://github.com/dl5di/OpenDV/pull/10
>
>
>         Commit Summary
>
>   * bugfix for gateway mode stripping slow data, not tested yet
>   * Still bugfixing for network originated slow data being stripped.
>   * Fixed a bug where an unhandled exception was thrown in
>     Slowdataencoder. Fixed a bug where slowdata coming from the
>     network was stripped when in gateway mode.
>
>
>         File Changes
>
>   * *M* DStarRepeater/Common/SlowDataEncoder.cpp
>     <https://github.com/dl5di/OpenDV/pull/10/files#diff-0> (2)
>   * *M* DStarRepeater/DStarRepeater/DStarRepeaterTRXThread.cpp
>     <https://github.com/dl5di/OpenDV/pull/10/files#diff-1> (4)
>
>
>         Patch Links:
>
>   * https://github.com/dl5di/OpenDV/pull/10.patch
>   * https://github.com/dl5di/OpenDV/pull/10.diff
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/dl5di/OpenDV/pull/10>.
>


-- 
Vy 73
Hans DL5DI

**Comment by F4FXL** [comment]
Hi Hans,

What repository is Jonathan using ?

I am not sure to understand his answer correctly let me rephrase it to see if I got it.

The slow data stripping code was added because somehow the slow data shall not make it through when the header is missed. Do I understand that correctly?

If the latter is correct I think this not acting as it should. I ran/listened to hours of QSOs and I never ever got any slow data. This implies that all headers were always missed which I do not think is the case.
Another possibility is that the code that determines if the header has been missed is buggy those leading to the slow data stripping code kicking in when it should not.

Beste Grüsse aus Frankreich,
Geoffrey F4FXL

**Comment by F4FXL** [comment]
Hi,

I will close this, I have a better fix.
